### PR TITLE
Adding info on the rust build toolchain and usage

### LINKED ISSUE #1: How to use this? There is no script.
I want to do some rename bios files from linux, but don't see how i can install this.

### PR DESCRIPTION
**Pull request by h454n5**
Awesome tool! 
I'm sure there are tons of non-developer folks out there who want to unbrick their ASUS systems and don't have a windows PC handy. Added some ELI5 notes for those people.

### CODE REVIEW

**Comment by iAmSomeone2** [comment]
Thank you! Thanks for the contribution too!

You're exactly right about why I wrote the program. My desktop PC only had Fedora Linux installed and my other machine is a MacBook. Wine did get the official tool to work, but I figured writing something native would be a good way to learn a little Rust :).

File: README.md [hunk 1]
```diff
@@ -3,11 +3,38 @@
 Cross-platform Rust implementation of Asus' Windows-only BIOS renamer utility.
 
 ## Usage
+Before you can use this, you need to compile the code to create an executable.
+
+### Install Rust on Linux/ Mac / etc
+```bash
+curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh -s -- --default-toolchain nightly
```
Comment (iAmSomeone2) [inline README.md]:
```suggestion
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```

Nightly definitely isn't necessary anymore, so let's use the officially recommended method.

File: README.md [hunk 2]
```diff
@@ -3,11 +3,38 @@
...
+```
+### You can even install rust with homebrew on MacOS
+```bash
+brew install rust 
+```
+
+### Confirm Rust Installed
+Once you install rust, you will have the build toolchain. You can confirm that by running these commands.
+``` cargo --version
+# cargo 1.65.0 (4bc8f24d3 2022-10-20)
```
Comment (iAmSomeone2) [inline README.md]:
```suggestion
# cargo 1.66.0 (d65d197ad 2022-11-15)
```
Anyone installing the toolchain now will be getting this version of Cargo.

File: README.md [hunk 3]
```diff
@@ -3,11 +3,38 @@
...
+brew install rust 
+```
+
+### Confirm Rust Installed
+Once you install rust, you will have the build toolchain. You can confirm that by running these commands.
+``` cargo --version
+# cargo 1.65.0 (4bc8f24d3 2022-10-20) 
+
+rustc --version
+# rustc 1.65.0
```
Comment (iAmSomeone2) [inline README.md]:
```suggestion
# rustc 1.66.0 (69f9c33d7 2022-12-12)
```

File: README.md [hunk 4]
```diff
@@ -21,3 +48,20 @@ the file.
...
     ```
 4. Follow the instructions given by the application.
+
+# Easy Script for non-developers:
+This will work on linux and mac
+
+```bash
+#!/bin/bash
+## install rust
+curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh -s -- --default-toolchain nightly
```
Comment (iAmSomeone2) [inline README.md]:
```suggestion
curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```

File: README.md [hunk 5]
```diff
@@ -3,11 +3,38 @@
 Cross-platform Rust implementation of Asus' Windows-only BIOS renamer utility.
 
 ## Usage
+Before you can use this, you need to compile the code to create an executable.
+
+### Install Rust on Linux/ Mac / etc
```
Comment (iAmSomeone2) [inline README.md]:
```suggestion
### Install Rust on Unix-like OSs: Linux, macOS, etc.
```
Just a small change to make it more clear which OSs are supported by this method.

File: README.md [hunk 6]
```diff
@@ -3,11 +3,38 @@
...
+```bash
+curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh -s -- --default-toolchain nightly
+```
+### You can even install rust with homebrew on MacOS
+```bash
+brew install rust 
+```
+
+### Confirm Rust Installed
```
Comment (iAmSomeone2) [inline README.md]:
```suggestion
### Confirm Rust is Installed
```

File: README.md [hunk 7]
```diff
@@ -3,11 +3,38 @@
...
+Once you install rust, you will have the build toolchain. You can confirm that by running these commands.
```
Comment (iAmSomeone2) [inline README.md]:
```suggestion
Once you install Rust, you will have the build toolchain. You can confirm that by running these commands.
```

**Review by iAmSomeone2 — CHANGES_REQUESTED** [review]
Thanks for the great work! I added some suggestions to, hopefully, make the new sections even clearer. I also noted where we should encourage using the current stable Rust toolchain. Considering that the target audience is hobbyist Linux desktop users, we should avoid asking people to use unstable tools where it's not necessary. That should help reduce friction and make diagnosing issues easier in the event that things don't go as smoothly as they should.

**Comment by h454n5** [comment]
Good stuff. I did notice the logic fails to get the proper asus flashback filename for several examples. Maybe parsing segments from the bios release may not reliably work. (Use case: any of the Crosshair x570 motherboards)

Maybe just mapping the models to the release will be easy enough. There aren't many. I would have done it, but can't find them all listed in one place. 

... or maybe even searching the entire release filename on Asus support site, then parsing the results page to capture the portion of the description that has the flashback filename listed.

**Review by iAmSomeone2 — APPROVED** [review]

**Comment by iAmSomeone2** [comment]
You're definitely right about the logic not working on all boards. I know for a fact that the Intel boards use a different naming scheme than the AMD AM4 boards, and each level (Strix, Crosshair, TUF, etc.) is different as well. The current implementation works for TUF AM4 boards since that's what I own and tested against.

Given that there's been some interest, I'll make some issues to add support for more boards. At the very least, we would need to:

1. Determine the name transformation patterns of the remaining boards.
2. Add logic to determine what kind of board the file is for
3. Apply specialized logic for detected board type
